src/tchannel_bridge.js

```diff
@@ -0,0 +1,163 @@
+    _tracer: Tracer;
+    _codec: TextMapCodec;
+
+    constructor(tracer: Tracer) {
+        this._tracer = tracer;
+        this._codec = new TextMapCodec({
+            urlEncoding: false,
+            contextKey: TCHANNEL_TRACING_PREFIX + constants.TRACER_STATE_HEADER_NAME,
+            baggagePrefix: TCHANNEL_TRACING_PREFIX + constants.TRACER_BAGGAGE_HEADER_PREFIX
+        });
+    }
+
+    _tchannelCallbackWrapper(span, callback, err, res) {
+        if (err) {
+            span.setTag(opentracing.Tags.ERROR, true);
+            span.log('error_msg', err);
+        }
+
+        span.finish();
+        return callback(err, res);
+    }
+
+    /**
+     * Wraps a tchannel handler, and takes a context in order to populate the incoming context
+     * with a span.
+     *
+     * @param {Function} [handlerFunc] - a tchannel handler function that responds to an incoming request.
+     * @param {Object} [options] - options to be passed to a span on creation.
+     * @returns {Function} - a function that wrapps the handler in order to automatically populate
+     * a the handler's context with a span.
+     **/
+    tracedHandler(handlerFunc: any, options: startSpanArgs = {}): Function {
+        return (context, request, headers, body, callback) => {
+            let operationName = options.operationName || request.arg1;
+            let span = this._extractSpan(operationName, headers);
+
+            // set tags
+            span.setTag(opentracing.Tags.PEER_SERVICE, request.callerName);
+            let hostPort = request.remoteAddr.split(':');
+            if (hostPort.length == 2) {
+                span.setTag(opentracing.Tags.PEER_HOST_IPV4, Utils.ipToInt(hostPort[0]));
+                span.setTag(opentracing.Tags.PEER_PORT, parseInt(hostPort[1]));
+            }
+            if (request.headers && request.headers.as) {
+                span.setTag('as', request.headers.as);
+            }
+
+            // In theory may overwrite tchannel span, but thats what we want anyway.
+            context.openTracingSpan = span;
+
+            // remove headers prefixed with $tracing$
+            for (let key in headers) {
+                if (headers.hasOwnProperty(key) && Utils.startsWith(key, TCHANNEL_TRACING_PREFIX)) {
+                    delete headers[key];
+                }
+            }
+
+            let wrappingCallback = this._tchannelCallbackWrapper.bind(null, span, callback);
+            handlerFunc(context, request, headers, body, wrappingCallback);
+        };
+    }
+
+    _wrapTChannelSend(wrappedSend, channel, req, endpoint, headers, body, callback) {
+        headers = headers || {};
+        let context = req.context || {};
+        // if opentracingContext.openTracingSpan is null, then start a new root span
+        // else start a span that is the child of the context span.
+        let childOf = context.openTracingSpan;
+        context.openTracingSpan = this._tracer.startSpan(endpoint, {
```

Comment:
you cannot replace span in the context with the child span, it means you will mess up the handler's span